src/fight_club_db_service.py

```diff
@@ -0,0 +1,97 @@
+from model import *
+import operator
+from functools import reduce
+
+class FightClubDbService:
+	fighter_parameters_map = {
+		'index': Fighter.index,
+		'name': Fighter.name,
+		'hp': Fighter.hp,
+		'damage_dice': Fighter.damage_dice,
+		'damage_bonus': Fighter.damage_bonus
+	}
+
+	fight_parameters_map = {
+		'winner': Fight.winner,
+		'rounds_number': Fight.rounds_number,
+		'ko': Fight.ko
+	}
+
+	round_parameters_map = {
+		'fight_id': Round.fight_id,
+		'round_nr': Round.round_nr,
+		'attacker': Round.attacker,
+		'defender': Round.defender,
+		'damage': Round.damage,
+		'previous_hp': Round.previous_hp,
+		'current_hp': Round.current_hp
+	}
+
+	@staticmethod
+	def find_fighter_by_index(index: str) -> 'Fighter':
+		fighter = Fighter.get_or_none(Fighter.index == index)
+		return fighter
+
+	@staticmethod
+	def save_fighter(fighter_index: str, dto: 'FighterDto'):
+		Fighter.create(index=fighter_index, name=dto.name, hp=dto.hp, damage_dice=dto.damage_dice, damage_bonus=dto.damage_bonus)
+    
+	@staticmethod
+	def save_fight_result(fight_result: 'FightResultDto'):
+		fight = Fight.create(winner=fight_result.victory.winner_name, rounds_number=fight_result.victory.rounds_number, ko=fight_result.victory.ko)
+		for round in fight_result.rounds:
+			Round.create(fight_id=fight, round_nr=round.round_number, attacker=round.attacker, defender=round.defender, damage=round.damage, previous_hp=round.previous_hp, current_hp=round.current_hp)
+
+	@staticmethod
+	def find_fighters_by_params(params):
+		predicates = []
+		for key, value in params:
+			if key not in FightClubDbService.fighter_parameters_map:
+				raise AttributeError(f"Parameter {key} is unknown for Fighter object."
```

Comment:
Please rise a custom exception here

Reply:
Why different if that one is intended to handle such cases?

Reply:
ok